pandas/core/arrays/integer.py

```diff
@@ -356,6 +356,16 @@ def __pos__(self):
     def __abs__(self):
         return type(self)(np.abs(self._data), self._mask)
 
+    @classmethod
+    def _from_scalars(cls, data, dtype):
+        # override because dtype.type is only the numpy scalar
+        # TODO accept float here?
+        if not all(
+            isinstance(v, (int, dtype.type, float, np.float_)) or isna(v) for v in data
```

Comment:
for floats require that `v.is_integer()`?

Reply:
The `coerce_to_arrray` function that is used by `_from_sequence` already checks for this as well (so we can pass through here any float, as it will be catched later)

Reply:
DTA/TDA/PA have a `_recognized_scalars` attribute that could be useful